pkg/transientstore/common/common_store_helper.go

```diff
@@ -0,0 +1,168 @@
+	var compositeKey []byte
+	compositeKey = append(compositeKey, []byte(txid)...)
+	compositeKey = append(compositeKey, compositeKeySep)
+	compositeKey = append(compositeKey, []byte(uuid)...)
+	compositeKey = append(compositeKey, compositeKeySep)
+	compositeKey = append(compositeKey, util.EncodeOrderPreservingVarUint64(blockHeight)...)
+
+	return compositeKey
+}
+
+// CreateCompositeKeyForPurgeIndexByHeight creates a key to index private write set based on
+// received at block height such that purge based on block height can be achieved. The structure
+// of the key is <purgeIndexByHeightPrefix>~blockHeight~txid~uuid.
+// TODO add pinning script to expose this function
+func CreateCompositeKeyForPurgeIndexByHeight(blockHeight uint64, txid string, uuid string) []byte {
+	var compositeKey []byte
+	compositeKey = append(compositeKey, purgeIndexByHeightPrefix)
+	compositeKey = append(compositeKey, compositeKeySep)
+	compositeKey = append(compositeKey, util.EncodeOrderPreservingVarUint64(blockHeight)...)
+	compositeKey = append(compositeKey, compositeKeySep)
+	compositeKey = append(compositeKey, []byte(txid)...)
+	compositeKey = append(compositeKey, compositeKeySep)
+	compositeKey = append(compositeKey, []byte(uuid)...)
+
+	return compositeKey
+}
+
+// SplitCompositeKeyOfPvtRWSet splits the compositeKey (<prwsetPrefix>~txid~uuid~blockHeight)
+// into uuid and blockHeight.
+// TODO add pinning script to expose this function
+func SplitCompositeKeyOfPvtRWSet(compositeKey []byte) (uuid string, blockHeight uint64) {
+	return splitCompositeKeyWithoutPrefixForTxid(compositeKey[2:])
+}
+
+// SplitCompositeKeyOfPurgeIndexByHeight splits the compositeKey (<purgeIndexByHeightPrefix>~blockHeight~txid~uuid)
+// into txid, uuid and blockHeight.
+// TODO add pinning script to expose this function
+func SplitCompositeKeyOfPurgeIndexByHeight(compositeKey []byte) (txid string, uuid string, blockHeight uint64) {
+	var n int
+	blockHeight, n = util.DecodeOrderPreservingVarUint64(compositeKey[2:])
+	splits := bytes.Split(compositeKey[n+3:], []byte{compositeKeySep})
+	txid = string(splits[0])
+	uuid = string(splits[1])
+	return
+}
+
+// splitCompositeKeyWithoutPrefixForTxid splits the composite key txid~uuid~blockHeight into
+// uuid and blockHeight
+func splitCompositeKeyWithoutPrefixForTxid(compositeKey []byte) (uuid string, blockHeight uint64) {
+	// skip txid as all functions which requires split of composite key already has it
+	firstSepIndex := bytes.IndexByte(compositeKey, compositeKeySep)
+	secondSepIndex := firstSepIndex + bytes.IndexByte(compositeKey[firstSepIndex+1:], compositeKeySep) + 1
+	uuid = string(compositeKey[firstSepIndex+1 : secondSepIndex])
+	blockHeight, _ = util.DecodeOrderPreservingVarUint64(compositeKey[secondSepIndex+1:])
+	return
+}
+
+// GetTransientStorePath returns the filesystem path for temporarily storing the private rwset
+// TODO add pinning script to keep this function
+func GetTransientStorePath() string {
```

Comment:
we don't need this function as this file was copied from Fabric, it can be removed

Reply:
done